doc/lightningd-config.5.txt

```diff
@@ -0,0 +1,302 @@
+
+The lightning-listconfigs(7) command will output a valid configuration
+file using the current settings.
+
+OPTIONS
+-------
+
+General options:
+
+*allow-deprecated-apis*='BOOL'::
+    Enable deprecated options, JSONRPC commands, fields, etc.  It
+    defaults to 'true', but you should set it to 'false' when testing
+    to ensure that an upgrade won't break your configuration.
+
+*help*::
+    Print help and exit.  Not very useful inside a configuration file, but
+    fun to put in other's config files while their computer is unattended.
+
+*version*::
+    Print version and exit.  Also useless inside a configuration file,
+    but putting this in someone's config file may convince them to
+    read this man page.
+
+Bitcoin control options:
+
+*network*='NETWORK'::
+    Select the network parameters ('bitcoin', 'testnet', or 'regtest').
+
+*testnet*::
+    Alias for 'network=testnet'.
+
+*mainnet*::
+    Alias for 'network=bitcoin'.
+
+*bitcoin-cli*='PATH'::
+    The name of 'bitcoin-cli' executable to run.
+
+*bitcoin-datadir*='DIR'::
+    '-datadir' argument to supply to bitcoin-cli(1).
+
+*bitcoin-rpcuser*='USER'::
+    The RPC username for talking to bitcoind(1).
+
+*bitcoin-rpcpassword*='PASSWORD'::
+    The RPC password for talking to bitcoind(1).
+
+*bitcoin-rpcconnect*='HOST'::
+    The bitcoind(1) RPC host to connect to.
+
+*bitcoin-rpcport*='PORT'::
+    The bitcoind(1) RPC port to connect to.
+
+*default-fee-rate*='SIPA'::
+    Satoshis per kw if bitcoind can't estimate fees.
+
+*rescan*='BLOCKS'::
+    Number of blocks to rescan from the current head, or absolute blockheight
+    if negative.  This is only needed if something goes badly wrong.
+
+Lightning daemon options:
+
+*lightning-dir*='DIR'::
+    Sets the working directory. All other files are relative to this.
+
+*pid-file*='PATH'::
+    Specify pid file to write to.
+  
+*log-level*='LEVEL'::
+    What log level to print out: options are io, debug, info, unusual, broken.
+
+*log-prefix*='PREFIX'::
+    Prefix for log lines: this can be customized if you want to merge logs with
+    multiple daemons.
+
+*log-file*='PATH'::
+    Log to this file instead of stdout.
+
+*rpc-file*='PATH'::
+    Set JSON-RPC socket (or /dev/tty), such as for lightning-cli(1).
+
+*daemon*::
+    Run in the background, suppress stdout and stderr.
+
+Lightning node customization options:
+
+*rgb*='RRGGBB'::
+    Your favorite color as a hex code.
+
+*alias*='NAME'::
+    Up to 32 UTF-8 characters to tag your node.  Completely silly, since anyone
+    can call their node anything they want.  The default is an
+    NSA-style codename derived from your public key, but "Peter Todd"
+    and "VAULTERO" are good options, too.
+
+*fee-base*='MILLISATOSHI'::
+    The base fee to charge for every payment which passes through.  Note that
+    millisatoshis are a very, very small unit!
+
+*fee-per-satoshi*='MILLIONTHS'::
+    This is the proportional fee to charge for every payment which passes
```

Comment:
`fee-base` and `fee-per-satoshi` would be better as something like `fee-per-transaction` and `fee-per-value` or similar, wouldn't they? `millisatoshis-per-kilosatoshi` would be a more accurate, but probably not helpful, description of the value of the latter field. I wonder if `fee-percent` and saying its denominated in 10,000ths of a percent mightn't be clearer though?

Reply:
All names have issues, AFAICT.   The spec calls them `fee_base_msat` and `fee_proportional_millionths`.